examples/cc3200/xively_firmware_updates/SFT/xi_sft_client.c

```diff
@@ -0,0 +1,1336 @@
+/*  Copyright (c) 2003-2017, LogMeIn, Inc. All rights reserved.
+    This is part of Xively C library. */
+
+
+/*
+ * HEADERS
+ */
+
+/* SFT Client Includes */
+#include <xively.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include "../cbor/cbor.h"
+#include "../cbor/cn-cbor/cn-cbor.h"
+#include "../sha256/sha256.h"
+#include "xi_sft.h"
+
+/* Simpelink EXT lib includes */
+#include "flc_api.h"
+
+/* Platform includes */
+#include "hw_memmap.h"
+#include "hw_common_reg.h"
+#include "hw_types.h"
+#include "hw_ints.h"
+#include "interrupt.h"
+#include "rom.h"
+#include "rom_map.h"
+#include "uart.h"
+#include "prcm.h"
+#include "utils.h"
+#include "gpio.h"
+#include "gpio_if.h"
+#include "uart_if.h"
+
+/*
+ * DEFINES
+ */
+#define UART_PRINT Report
+#undef printf
+#define printf Report
+
+#define MAX_INCOMING_FILENAME_LENGTH 50
+#define MAX_STRING_LENGTH 50
+#define MESSAGESIZE 50
+#define LOGDETAILSSIZE 400
+
+/*
+ * TYPEDEFS
+ */
+typedef unsigned char uchar;
+typedef unsigned int uint;
+
+typedef enum {
+    sft_status_ok = 0, /* 0 */
+
+    sft_status_rebooting, /* 1 */
```

Comment:
Yeah, I think it's less about reading the value and instead about using it as an index into a string table for the device logs.  If there's a typo error instead of an autoassigned value, then the error will be mislogged.  I can make them assignments and be careful about it.